### DISCUSSION TITLE
2024.9.0

### DISCUSSION THREAD

**Discussion by AndrewBastin** [hoppscotch/hoppscotch#4400] [Announcements]
This is a major release and introduces a bunch of new features like the new Hoppscotch Agent Interceptor, JSON with Comments support in request body, API Request Response Examples along with fixes for the desktop app, SVG rendering and importers.

## What's Changed
* feat: add loading state for import actions by @HelixY2J in https://github.com/hoppscotch/hoppscotch/pull/4217
* refactor: make global environment a versioned entity by @ipalakchopra in https://github.com/hoppscotch/hoppscotch/pull/4216
* fix(desktop): content security policy config preventing app launch by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4365
* feat: duplicate global environment under team workspaces by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4334
* chore: addition of new GQL subscription for `UserCollection` duplication by @balub in https://github.com/hoppscotch/hoppscotch/pull/4358
* feat: add generate data schema option in spotlight searcher by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4356
* chore: export single User/Team Collection by @balub in https://github.com/hoppscotch/hoppscotch/pull/4359
* chore: make client secret optional across grant types by @amk-dev in https://github.com/hoppscotch/hoppscotch/pull/4363
* feat: import to the current workspace from any team / personal workspace by @amk-dev in https://github.com/hoppscotch/hoppscotch/pull/4377
* feat: save api response by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4382
* feat: extend duplicate collection to personal workspace in SH by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4368
* feat: modify body with ai & feedback on ai requests hoppscotch-common bindings by @amk-dev in https://github.com/hoppscotch/hoppscotch/pull/4386
* feat: support for comments on `application/json` request body by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4335
* fix: request variable list item remove bug by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4392
* fix: update copy + improve ux on ai experiments by @amk-dev in https://github.com/hoppscotch/hoppscotch/pull/4393
* fix: add fallback for `type` field in `restTabState` schema by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4398
* fix: prettier for json with comments by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4399
* fix: correct SVG rendering issue by adjusting blob type (#3565) by @shuaixr in https://github.com/hoppscotch/hoppscotch/pull/4397
* feat: hoppscotch agent and agent interceptor  by @AndrewBastin in https://github.com/hoppscotch/hoppscotch/pull/4396

## New Contributors
* @HelixY2J made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4217
* @ipalakchopra made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4216
* @shuaixr made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4397

**Full Changelog**: https://github.com/hoppscotch/hoppscotch/compare/2024.8.3...2024.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/hoppscotch/hoppscotch/releases/tag/2024.9.0'>2024.9.0</a>.</em>

**Comment by girish2711** [comment]
Latest release does not pick up the chosen environments. No matter how many times I choose and set environment, variables are not resolved.

**Reply by jamesgeorge007** [reply]
Hi, sorry for the inconvenience. Can you open an issue with the steps to reproduce?

**Reply by slogivewd** [reply]
Same here, can use any env

**Reply by jamesgeorge007** [reply]
Hi, we have a [patch release](https://github.com/hoppscotch/hoppscotch/releases/tag/2024.9.1) out. Please let us know if the issue persists. You can perform the upgrade directly from the Desktop app via the upgrade prompt or download the latest version [here](https://hoppscotch.com/download).
